src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.netcoreapp.cs

```diff
@@ -0,0 +1,476 @@
+
+                ExpectEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryOne), cleanup: null, expectedPath: directoryOne);
+                watcher.Created -= createHandler;
+                ExpectEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryTwo), cleanup: null, expectedPath: directoryTwo);
+                watcher.Created -= createHandler;
+                ExpectNoEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryThree), cleanup: null, expectedPath: directoryThree);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_Directory_Create_Filter_Ctor()
+        {
+            // Check create events against multiple filters
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            string directoryOne = Path.Combine(directory.FullName, GetTestFileName());
+            string directoryTwo = Path.Combine(directory.FullName, GetTestFileName());
+            string directoryThree = Path.Combine(directory.FullName, GetTestFileName());
+
+            using (var watcher = new FileSystemWatcher(directory.FullName, Path.GetFileName(directoryOne)))
+            {
+                watcher.Filters.Add(Path.GetFileName(directoryTwo));
+
+                ExpectEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryOne), cleanup: null, expectedPath: directoryOne);
+                watcher.Created -= createHandler;
+                ExpectEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryTwo), cleanup: null, expectedPath: directoryTwo);
+                watcher.Created -= createHandler;
+                ExpectNoEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryThree), cleanup: null, expectedPath: directoryThree);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_Directory_Delete_MultipleFilters()
+        {
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            DirectoryInfo directoryOne = Directory.CreateDirectory(Path.Combine(directory.FullName, GetTestFileName()));
+            DirectoryInfo directoryTwo = Directory.CreateDirectory(Path.Combine(directory.FullName, GetTestFileName()));
+            DirectoryInfo directoryThree = Directory.CreateDirectory(Path.Combine(directory.FullName, GetTestFileName()));
+
+            using (var watcher = new FileSystemWatcher(directory.FullName))
+            {
+                watcher.Filters.Add(Path.GetFileName(directoryOne.FullName));
+                watcher.Filters.Add(Path.GetFileName(directoryTwo.FullName));
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, () => directoryOne.Delete(), cleanup: null, expectedPath: directoryOne.FullName);
+                watcher.Deleted -= deleteHandler;
```

Comment:
It doesn't make sense to unhook here. Move it where you have the context (i.e. `ExecuteAndVerifyEvents`).